visualization/app/codeCharta/util/algorithm/treeMapLayout/treeMapHelper.ts

```diff
@@ -203,13 +204,73 @@ export function getBuildingColor(node: CodeMapNode, { appSettings, dynamicSettin
 		return mapColors.flat
 	}
 
-	if (metricValue < dynamicSettings.colorRange.from) {
-		return mapColors.positive
-	}
-	if (metricValue > dynamicSettings.colorRange.to) {
-		return mapColors.negative
+	const positiveColorRGB = ColorConverter.convertHexToColorObject(mapColors.positive)
+	const neutralColorRGB = ColorConverter.convertHexToColorObject(mapColors.neutral)
+	const negativeColorRGB = ColorConverter.convertHexToColorObject(mapColors.negative)
+
+	switch (dynamicSettings.colorMode) {
```

Comment:
We should consider to outsource each case into separate functions and to call them in the switch/case accordingly.
And I think the last case can be ommitted because it would be the default "return".